seabird: init at 0.2.2

nicolas-goudry opened the pull request:
## Description of changes

<!--
For package updates please link to a changelog or describe changes, this helps your fellow maintainers discover breaking updates.
For new packages please briefly describe the package or provide a link to its homepage.
-->

Graphical Kubernetes desktop client: https://getseabird.github.io

## Things done

<!-- Please check what applies. Note that these are not hard requirements but merely serve as information for reviewers. -->

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [x] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [24.05 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2405.section.md) (or backporting [23.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2305.section.md) and [23.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

<!--
To help with the large amounts of pull requests, we would appreciate your
reviews of other pull requests, especially simple package updates. Just leave a
comment describing what you have tested in the relevant package/service.
Reviewing helps to reduce the average time-to-merge for everyone.
Thanks a lot if you do!

List of open PRs: https://github.com/NixOS/nixpkgs/pulls
Reviewing guidelines: https://nixos.org/manual/nixpkgs/unstable/#chap-reviewing-contributions
-->

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

nicolas-goudry committed:
seabird: init at 0.2.1

OmnipotentEntity reviewed (approved):
Diff looks good, pr review: builds and launches executable.

knownasred reviewed (approved):
LGTM

Builds on x86_64-linux successfully, and the program launches & behaves as expected (I managed to access a test cluster without issues)

nicolas-goudry commented:
Updated PR to init at 0.2.1 (just released).

```diff
diff --git a/pkgs/by-name/se/seabird/package.nix b/pkgs/by-name/se/seabird/package.nix
index a3fab66d59f4..fcf5175e3aeb 100644
--- a/pkgs/by-name/se/seabird/package.nix
+++ b/pkgs/by-name/se/seabird/package.nix
@@ -15,16 +15,16 @@
 
 buildGo122Module rec {
   pname = "seabird";
-  version = "0.1.2";
+  version = "0.2.1";
 
   src = fetchFromGitHub {
     owner = "getseabird";
     repo = "seabird";
     rev = "v${version}";
-    hash = "sha256-jPdAKdRHdtxCzgv5nIf3FUyGS+0tOz8idg2B19uX5cg=";
+    hash = "sha256-wEZS5wOiSaCJ5jEx5tfp6QxY0SgwP5e7MV+iqekHmPY=";
   };
 
-  vendorHash = "sha256-n/KQAThQ8aft5dlHefMZxK2/EUnfXXJfynqHNiBZrnE=";
+  vendorHash = "sha256-z9l6g5NkAErRQo8oiqwKG9ssm8K2S+eSZBD0w4kO3kc=";
 
   nativeBuildInputs = [
     copyDesktopItems
```

@OmnipotentEntity @valentin-ricard Would you be kind enough to test this again? Thanks!

knownasred reviewed (approved):
Builds and works as expected.

But I have taken a look at the "About" section, but "dev" is set to the version number.

I've taken a quick look at the actions they use to release a version, and it seems they make a patch to set the version in the app: https://github.com/getseabird/seabird/blob/1f1f4dbbeeb3a373032b4d6ddcafd6aa8b974e20/.github/workflows/release.yaml#L47C1-L47C62

Is that a comportment we want to replicate?

---
1 package built:
seabird

Result of `nixpkgs-review pr 292581` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>1 package built:</summary>
  <ul>
    <li>seabird</li>
  </ul>
</details>

nicolas-goudry commented:
> But I have taken a look at the "About" section, but "dev" is set to the version number.

Nice catch, it’s now fixed.

knownasred reviewed (approved):
![Screenshot from 2024-03-03 19-46-46](https://github.com/NixOS/nixpkgs/assets/91870700/fcb11083-20fb-401f-9216-90fc83b711ed)

LGTM with the most recent fix :)

---

Result of `nixpkgs-review pr 292581` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>1 package built:</summary>
  <ul>
    <li>seabird</li>
  </ul>
</details>

SuperSandro2000 commented on pkgs/by-name/se/seabird/package.nix:
```diff
@@ -0,0 +1,72 @@
+
+  nativeBuildInputs = [
+    copyDesktopItems
+    glib
```
I don' think glib is used

nicolas-goudry replied:
I added it because of what `ldd` output:

```
$ ldd /nix/store/bnbnq9sgzpnimwj3k1vixjdwh9a8l2hg-seabird-0.2.1/bin/seabird
	linux-vdso.so.1 (0x00007ffdb2bc2000)
	libc.so.6 => /nix/store/8mc30d49ghc8m5z96yz39srlhg5s9sjj-glibc-2.38-44/lib/libc.so.6 (0x00007f0108c59000)
	/nix/store/8mc30d49ghc8m5z96yz39srlhg5s9sjj-glibc-2.38-44/lib/ld-linux-x86-64.so.2 => /nix/store/8mc30d49ghc8m5z96yz39srlhg5s9sjj-glibc-2.38-44/lib64/ld-linux-x86-64.so.2 (0x00007f0108e44000)

$ nix shell 'nixpkgs#nix-index' -c nix-locate libc.so.6 | grep -v '('
remarkable-toolchain.out                              0 s /nix/store/d69721qdx5y5iysfg5q1mnc5c4bh66ni-remarkable-toolchain-3.1.2/sysroots/cortexa9hf-neon-remarkable-linux-gnueabi/lib/libc.so.6
remarkable-toolchain.out                              0 s /nix/store/d69721qdx5y5iysfg5q1mnc5c4bh66ni-remarkable-toolchain-3.1.2/sysroots/x86_64-codexsdk-linux/lib/libc.so.6
remarkable2-toolchain.out                             0 s /nix/store/y59nachj812qs309ikz8gw6dx78m4gvv-remarkable2-toolchain-3.1.2/sysroots/cortexa7hf-neon-remarkable-linux-gnueabi/lib/libc.so.6
remarkable2-toolchain.out                             0 s /nix/store/y59nachj812qs309ikz8gw6dx78m4gvv-remarkable2-toolchain-3.1.2/sysroots/x86_64-codexsdk-linux/lib/libc.so.6
glibc_memusage.debug                                  0 s /nix/store/km9ggz0h354f3vldgzvgjjyiblvw8hn5-glibc-gd-2.37-45-debug/lib/debug/libc.so.6
glibc_memusage.out                            2,252,736 x /nix/store/y51mzms9hyaz3khsmahvwp1i5n9jfpp3-glibc-gd-2.37-45/lib/libc.so.6
glibc.out                                     2,252,736 x /nix/store/whypqfa83z4bsn43n4byvmw80n4mg3r8-glibc-2.37-45/lib/libc.so.6
glibc_multi.out                                       0 s /nix/store/pq23w2wmy4kzndyv06c1a6k0mkpqhff4-glibc-multi-2.37-45/lib/libc.so.6
glibc.debug                                           0 s /nix/store/fd22xxhwvz51drm2i3yzzbyj1b363mgp-glibc-2.37-45-debug/lib/debug/libc.so.6
```

I’m still confused where to put dependencies, maybe it should go in `buildInputs` instead?

wegank replied:
`glib` and `glibc` are very different, though.

nicolas-goudry replied:
:facepalm: What a dummy… Thanks for pointing this out!

nicolas-goudry commented:
Updated PR to init at 0.2.2 (released last week). Removed `glib` from `nativeBuildInputs`. Added `glibc` to `buildInputs`.

---

1 package built:
seabird

Result of `nixpkgs-review pr 292581` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>1 package built:</summary>
  <ul>
    <li>seabird</li>
  </ul>
</details>

wegank commented on pkgs/by-name/se/seabird/package.nix:
```diff
@@ -0,0 +1,72 @@
+    go generate internal/icon/icon.go
+  '';
+
+  install = ''
```
`installPhase` or `postInstall`?

nicolas-goudry replied:
Fixed, I must have been drunk when I wrote this recipe…

wegank commented on pkgs/by-name/se/seabird/package.nix:
```diff
@@ -0,0 +1,72 @@
+  ];
+
+  buildInputs = [
+    glibc
```
```suggestion
```

`glibc` being the standard C library on Linux, you'll expect it to be implicitly included in any derivation requiring compilations.

nicolas-goudry replied:
Noted, I’ll remove it.

nicolas-goudry commented:
Replaced (invalid) `install` phase by `postInstall` hook. Also fixed invalid icon location.

---

Result of `nixpkgs-review pr 292581` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>1 package built:</summary>
  <ul>
    <li>seabird</li>
  </ul>
</details>

wegank commented:
Result of `nixpkgs-review pr 292581` run on aarch64-darwin [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>1 package built:</summary>
  <ul>
    <li>seabird</li>
  </ul>
</details>
